Automattic/woocommerce-payments
# Add filter for payment request button payment method title

#### Changes proposed in this Pull Request


# Add filter to replace "(WooCommerce Payments)" suffix from Google/Apple Pay payment methods

In #2434 it was decided that we should remove the mention of "(WooCommerce Payments)" from the suffix of UPE payment methods. At the time, since Apple and Google Pay differentiated themselves by being consumer brands, we decided to leave the suffix on those payment request payment methods, when #2434 was resolved in #2435.

However, due to [popular demand from merchants](https://github.com/Automattic/woocommerce-payments/pull/2435#issuecomment-992685127), in the case of payment request button payment methods, we decided to give merchants the option to replace this suffix with a filter. This follows the [example pioneered on the WC Stripe plugin](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/44f8f59a98b0fe8502008c28be8ab087e866723d/includes/payment-methods/class-wc-stripe-payment-request.php#L83).

This PR adds the filter `wcpay_payment_request_payment_method_title_suffix`, which can be used to replace the suffix on the displayed payment method title after checkout for payment request buttons, such as Apple and Google Pay.




#### Testing instructions





1. Pull the changes in this PR. To test Google Pay you will need to use Google Chrome and to test Apple Pay you will need to use Safari. Ensure that your site is running in Test Mode (if not, you will be charged for purchases) and communicating over HTTPS.
2. Enable Express Payment methods from **Payments > Settings**.
3. Purchase a product using, selecting an Express Payment method, wherever you enabled them in the previous step. 
4. After checkout, notice that the payment method title on the Order Received page ends with the string "(WooCommerce Payments)".
5. Add a filter on `wcpay_payment_request_payment_method_title_suffix` to change this suffix to an empty string or a value that you desire.
6. Repeat step 3. Notice that the payment method title on the Order Received page now reflects the alteration you have made using the filter in step 5.
7. Lucky number seven.

-------------------


**Post merge**
